src/ArticleList.jsx

```diff
@@ -2,15 +2,17 @@ import { useState, useEffect } from "react";
 import { fetchArticles } from "./api";
 import { ArticleCard } from "./ArticleCard";
 import "./App.css";
+import { useParams } from "react-router";
 
 export const ArticleList = () => {
+  const { topic } = useParams()
   const [articles, setArticles] = useState([]);
 
   useEffect(() => {
-    fetchArticles().then((articles) => {
+    fetchArticles(topic).then((articles) => {
       setArticles(articles);
     });
-  }, []);
+  }, [topic]);
```

Comment:
Your page took a while to load and this is normal as Render can be slow. But as a user, I just saw a blank page with a header. I would definitely add a loading state in here to give the user feedback that your page is working and articles are coming!